### PR TITLE
Add parameters dedicated to Dynaflow (#118)

### LINKED ISSUE #118: [DynaFlow] Additional parameters
* **Do you want to request a *feature* or report a *bug*?**

Feature

* **What is the expected behavior?**

When running dynaflow, we should be able to configure the following parameters (as they should appear in a config.json as input to dynaflow):

```json
    "ChosenOutputs" : ["STEADYSTATE", "LOSTEQ", "TIMELINE", "CONSTRAINTS"],
    "VSCAsGenerators": "true",
    "LCCAsLoads": "true",
    "TimeStep": 2.6,
```

### PR DESCRIPTION
**Pull request by dimbdr**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
Fixes #118 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Add new specific parameters to the Dynaflow API

**What is the current behavior?** *(You can also link to an open issue here)*
Currently, the Dynaflow API gives only 4 specific parameters.

**What is the new behavior (if this is a feature change)?**
Adding 4 more parameters to the Dynaflow API, for a total of 8 specific parameters.
Some were described in the PowSyBl documentation but were not available in the API. Documentation will be updated according to the current modifications.

### CODE REVIEW

File: dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java [hunk 1]
```diff
@@ -120,11 +188,14 @@ public static DynaFlowParameters load(Map<String, String> properties) {
         Optional.ofNullable(properties.get(SVC_REGULATION_ON)).ifPresent(prop -> parameters.setSvcRegulationOn(Boolean.parseBoolean(prop)));
         Optional.ofNullable(properties.get(SHUNT_REGULATION_ON)).ifPresent(prop -> parameters.setShuntRegulationOn(Boolean.parseBoolean(prop)));
         Optional.ofNullable(properties.get(AUTOMATIC_SLACK_BUS_ON)).ifPresent(prop -> parameters.setAutomaticSlackBusOn(Boolean.parseBoolean(prop)));
-        Optional.ofNullable(properties.get(DSO_VOLTAGE_LEVEL)).ifPresent(prop -> parameters.setDsoVoltageLevel(Double.parseDouble(prop)));
```
Comment (sylvlecl) [inline dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java:123]: We need to implement  the loading of new parameters here too.

Maybe code could be share with loading from the module ?

Comment (sylvlecl) [reply dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java:123]: Note that we should also implement the method for update from a map, in DynaFlowProvider, not only the creation from a map.

File: dynaflow/src/test/java/com/powsybl/dynaflow/DynaFlowParametersTest.java [hunk 2]
```diff
@@ -51,18 +55,31 @@ public void checkParameters() {
         boolean shuntRegulationOn = false;
         boolean automaticSlackBusOn = true;
         double dsoVoltageLevel = 87.32;
+        List<String> chosenOutputs = Collections.singletonList(OutputTypes.STEADYSTATE.name());
+        boolean vscAsGenerators = true;
+        boolean lccAsLoads = true;
+        double timeStep = 2.6;
```
Comment (sylvlecl) [inline dynaflow/src/test/java/com/powsybl/dynaflow/DynaFlowParametersTest.java:61]: You are testing the same values as default values, so it's not really checking that those values are taken into account

**Review by sylvlecl — CHANGES_REQUESTED** [review]
Thanks!
Implementation of loading from map still needs to be implemented :)

Note that we may implement at the same time that feature ? (but could be in another PR)
https://github.com/powsybl/powsybl-dynawo/issues/104#issuecomment-1262209738

File: dynaflow/src/test/java/com/powsybl/dynaflow/DynaFlowParametersTest.java (hunk 2 above)
Comment (dimbdr) [reply dynaflow/src/test/java/com/powsybl/dynaflow/DynaFlowParametersTest.java]: Indeed, modification done

File: dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java (hunk 1 above)
Comment (dimbdr) [reply dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java:123]: > We need to implement the loading of new parameters here too.
> 
> Maybe code could be shared with loading from the module ?

I made a little change, by providing a new signature called `loading`.
The `load` method is not broken, and I can use the method `loading` with the new implemented method `updateSpecificParameters` in the DynaFlowProvider.
Though, I'm not quite sure about the way I implemented this `updateSpecificParameters`.

**Review by sylvlecl — APPROVED** [review]
Ok for me, thank you for the update!
I added very minor comments, nothing blocking though

File: dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java [hunk 3]
```diff
@@ -111,20 +179,40 @@ private static void load(DynaFlowParameters parameters, ModuleConfig config) {
...
-                .setDsoVoltageLevel(config.getDoubleProperty(DSO_VOLTAGE_LEVEL, DEFAULT_DSO_VOLTAGE_LEVEL));
+                .setDsoVoltageLevel(config.getDoubleProperty(DSO_VOLTAGE_LEVEL, DEFAULT_DSO_VOLTAGE_LEVEL))
+                .setChosenOutputs(config.getStringListProperty(CHOSEN_OUTPUTS, DEFAULT_CHOSEN_OUTPUT))
+                .setVscAsGenerators(config.getBooleanProperty(VSC_AS_GENERATORS, DEFAULT_VSC_AS_GENERATORS))
+                .setLccAsLoads(config.getBooleanProperty(LCC_AS_LOADS, DEFAULT_LCC_AS_LOADS))
+                .setTimeStep(config.getDoubleProperty(TIME_STEP, DEFAULT_TIME_STEP));
     }
 
-    public static DynaFlowParameters load(Map<String, String> properties) {
+    public void loading(Map<String, String> properties) {
```
[the remark this answers is not in the document]
Comment (dimbdr) [reply dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java]: Agreed, changed to `update`

File: dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java [hunk 4]
```diff
@@ -111,20 +179,40 @@ private static void load(DynaFlowParameters parameters, ModuleConfig config) {
...
         Objects.requireNonNull(properties);
+        Optional.ofNullable(properties.get(SVC_REGULATION_ON)).ifPresent(prop -> setSvcRegulationOn(Boolean.parseBoolean(prop)));
+        Optional.ofNullable(properties.get(SHUNT_REGULATION_ON)).ifPresent(prop -> setShuntRegulationOn(Boolean.parseBoolean(prop)));
+        Optional.ofNullable(properties.get(AUTOMATIC_SLACK_BUS_ON)).ifPresent(prop -> setAutomaticSlackBusOn(Boolean.parseBoolean(prop)));
+        Optional.ofNullable(properties.get(DSO_VOLTAGE_LEVEL)).ifPresent(prop -> setDsoVoltageLevel(Double.parseDouble(prop)));
+        Optional.ofNullable(properties.get(CHOSEN_OUTPUTS)).ifPresent(prop ->
+                setChosenOutputs(Stream.of(prop.replace("[", "")
```
[the remark this answers is not in the document]
Comment (dimbdr) [reply dynaflow/src/main/java/com/powsybl/dynaflow/DynaFlowParameters.java]: Thanks for the info, I wasn't sure about the format.
Modified, assumed there's no bracket in the string

**Pull request merged**